tests/system/src/test/java/test/robot/javafx/scene/SliderTooltipNPETest.java

```diff
@@ -0,0 +1,158 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package test.robot.javafx.scene;
+
+import java.util.concurrent.CountDownLatch;
+
+import javafx.application.Application;
+import javafx.scene.input.MouseButton;
+import javafx.application.Platform;
+import javafx.scene.robot.Robot;
+import javafx.scene.Scene;
+import javafx.scene.control.Slider;
+import javafx.stage.Stage;
+import javafx.stage.StageStyle;
+import javafx.scene.control.Tooltip;
+import javafx.stage.WindowEvent;
+
+import org.junit.AfterClass;
+import org.junit.Assert;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import test.util.Util;
+
+/*
+ * Test for verifying Slider NPE error.
+ *
+ * There is 1 test in this file.
+ * Steps for testSliderTooltipNPE()
+ * 1. Create a slider and tooltip.
+ * 2. Make setAutoHide of tooltip as true and add tooltip to slider.
+ * 3. Hover over slider thumb and wait for tooltip. Drag the thumb.
+ * 4. Verify that NullPointerException is not thrown.
+ */
+
+public class SliderTooltipNPETest {
+    static CountDownLatch startupLatch = new CountDownLatch(1);
+    static CountDownLatch tooltipLatch = new CountDownLatch(1);
+    static Robot robot;
+    static Slider slider;
+
+    static volatile Throwable exception;
+    static volatile Stage stage;
+    static volatile Scene scene;
+
+    static final int SCENE_WIDTH = 250;
+    static final int SCENE_HEIGHT = SCENE_WIDTH;
+    static final int DRAG_DISTANCE = 10;
+
+    @Test
+    public void testSliderTooltipNPE() throws Throwable {
+        Assert.assertTrue(slider.getTooltip().getConsumeAutoHidingEvents());
+        dragSliderAfterTooltipDisplayed(DRAG_DISTANCE);
+        if (exception != null) {
+            exception.printStackTrace();
+            throw exception;
+        }
+        Assert.assertTrue(slider.getTooltip().getConsumeAutoHidingEvents());
+    }
+
+    private void dragSliderAfterTooltipDisplayed(int dragDistance) throws Exception {
+        Thread.sleep(1000); // Wait for slider to layout
```

Comment:
is there an equivalent of waitForIdle()?

could we replace the sleep() with waiting for some event/property instead?

Reply:
Here I followed the same pattern used in other system tests such as TabPaneDragPolicyTest and ColorPickerTest.